pyschema/source_generation.py

```diff
@@ -22,11 +25,124 @@ def to_python_source(classes, indent=DEFAULT_INDENT):
+        output_dir = os.path.dirname(output_file)
+        if not os.path.exists(output_dir):
+            os.makedirs(output_dir)
+
+        with open(output_file, 'w') as out_fn:
+            out_fn.write(module_code)
+
+    def write_init_files(self):
+        def touch_init_file_in_folder(folder):
+            path = os.path.join(folder, '__init__.py')
+            if not os.path.exists(path):
+                open(path, 'w').close()
+
+        touch_init_file_in_folder(self.target_folder)
+        for root, dirs, _ in os.walk(self.target_folder):
+            for d in dirs:
+                touch_init_file_in_folder(os.path.join(root, d))
+
+    def format_imports(self, imported_classes):
+        if not imported_classes:
+            return "\n"
+        imported_namespaces = self.get_namespace_clusters(imported_classes)
+        lines = []
+        for namespace, schemas in imported_namespaces.iteritems():
+            if self.parent_package:
+                if namespace:
+                    module = "{}.{}".format(self.parent_package, namespace)
+                else:
+                    module = self.parent_package
+            else:
+                if not namespace:
+                    module = "."
+                else:
+                    module = namespace
+            class_part = ", ".join(s.__name__ for s in schemas)
+            lines.append("from {} import {}".format(module, class_part))
+        return "\n".join(lines) + "\n\n"
+
+    def from_classes_with_refs(self, classes):
+        class_graph = CachedGraphTraverser()
+
+        all_classes = set(classes)
+        for c in classes:
+            referenced_schemas = class_graph.find_descendants(c)
+            all_classes |= set(referenced_schemas)
+
+        namespace_cluster = self.get_namespace_clusters(all_classes)
+
+        ordered_schemas = class_graph.get_reference_ordered_schemas(all_classes)
+
+        # Since we don't want to use the previous cached results we create a new instance
+        one_stack_level_class_graph = CachedGraphTraverser()
+        for namespace, classes in namespace_cluster.iteritems():
+            inlined_classes = [c for c in ordered_schemas if c in classes]
+            imported_classes = set()
+
+            for inlined in inlined_classes:
+                direct_references = child_only_class_graph.find_descendants(inlined, max_depth=1)
+                imported_classes |= set([c for c in direct_references if c not in inlined_classes])
+
+            module_code = (
+                header_source() +
+                self.format_imports(imported_classes) +
+                self.format_definitions(inlined_classes)
+            )
+            self.write_namespace_file(namespace, module_code)
+            self.write_init_files()
+
+
+def to_python_package(classes, target_folder, parent_package=None, indent=DEFAULT_INDENT):
+    '''
+    This function can be used to build a python package representation of pyschema classes.
```

Comment:
Would scrap "This function can be" part of the first sentence. Otherwise great docstring!